src/Auctions/Auction.sol

```diff
@@ -0,0 +1,571 @@
+        } else {
+            // Else just use current balance.
+            available = ERC20(_fromToken).balanceOf(address(this));
+        }
+
+        require(available != 0, "nothing to kick");
+
+        // Update the auctions status.
+        auctions[_auctionId].kicked = uint96(block.timestamp);
+        auctions[_auctionId].initialAvailable = available;
+        auctions[_auctionId].currentAvailable = available;
+
+        emit AuctionKicked(_auctionId, available);
+    }
+
+    /**
+     * @notice Take the token being sold in a live auction.
+     * @dev Defaults to taking the full amount and sending to the msg sender.
+     * @param _auctionId The unique identifier of the auction.
+     * @return . The amount of fromToken taken in the auction.
+     */
+    function take(bytes32 _auctionId) external virtual returns (uint256) {
+        return _take(_auctionId, type(uint256).max, msg.sender, new bytes(0));
+    }
+
+    /**
+     * @notice Take the token being sold in a live auction with a specified maximum amount.
+     * @dev Uses the sender's address as the receiver.
+     * @param _auctionId The unique identifier of the auction.
+     * @param _maxAmount The maximum amount of fromToken to take in the auction.
+     * @return . The amount of fromToken taken in the auction.
+     */
+    function take(
+        bytes32 _auctionId,
+        uint256 _maxAmount
+    ) external virtual returns (uint256) {
+        return _take(_auctionId, _maxAmount, msg.sender, new bytes(0));
+    }
+
+    /**
+     * @notice Take the token being sold in a live auction.
+     * @param _auctionId The unique identifier of the auction.
+     * @param _maxAmount The maximum amount of fromToken to take in the auction.
+     * @param _receiver The address that will receive the fromToken.
+     * @return _amountTaken The amount of fromToken taken in the auction.
+     */
+    function take(
+        bytes32 _auctionId,
+        uint256 _maxAmount,
+        address _receiver
+    ) external virtual returns (uint256) {
+        return _take(_auctionId, _maxAmount, _receiver, new bytes(0));
+    }
+
+    /**
+     * @notice Take the token being sold in a live auction.
+     * @param _auctionId The unique identifier of the auction.
+     * @param _maxAmount The maximum amount of fromToken to take in the auction.
+     * @param _receiver The address that will receive the fromToken.
+     * @param _data The data signify the callback should be used and sent with it.
+     * @return _amountTaken The amount of fromToken taken in the auction.
+     */
+    function take(
+        bytes32 _auctionId,
+        uint256 _maxAmount,
+        address _receiver,
+        bytes calldata _data
+    ) external virtual returns (uint256) {
+        return _take(_auctionId, _maxAmount, _receiver, _data);
+    }
+
+    /// @dev Implements the take of the auction.
+    function _take(
+        bytes32 _auctionId,
+        uint256 _maxAmount,
```

Comment:
Should you check for `maxAmount == 0`

Reply:
im okay with people sending  as a parameter paying slightly more gas to revert later so correct users pay less